typescript-client/api-reference/api-reference.mdx

```diff
@@ -66,7 +66,61 @@ The unique identifier of the step to delete.
 
 A `Promise` that resolves to the ID of the deleted step.
 
-### uploadFile()
+### createAttachment()
+
+> **createAttachment**(`params`): `Promise`\<`object`\>
+
+Uploads a file to Literal AI, and creates an attachment out of it. If called within a step wrapper, the attachment will be automatically added to the step.
+
+This method supports uploading either through direct content or via a file path. It first signs the upload through a pre-configured endpoint and then proceeds to upload the file using the signed URL.
+
+#### Parameters
+
+▪ **params**: `object`
+
+The parameters for uploading a file, including:
```

Comment:
Not the purpose for this PR, but ideally users see a summary JSON object like this:
```
{
    content: ReadableStream | ReadStream ...,
    path: Maybe<string>
}
```

Either that or some clearer indentation.

Reply:
Like that actually: 
![image](https://github.com/user-attachments/assets/c219548a-87fe-490f-ba9b-ee8f9b35756b)

Reply:
You're right i'm not a fan of this presentation either. As a whole this API reference feels very unsatisfying, it is just a list with no added information. I'll try to overhaul it when i find a minute.

Reply:
Plan is to focus on doc during August apparently.